app/src/main/java/com/x8bit/bitwarden/data/vault/repository/VaultRepositoryImpl.kt

```diff
@@ -553,6 +553,24 @@ class VaultRepositoryImpl(
         mutableTotpCodeResultFlow.tryEmit(totpCodeResult)
     }
 
+    override suspend fun unlockVaultWithAuthenticatorSyncKey(): VaultUnlockResult {
+        val userId = activeUserId ?: return VaultUnlockResult.InvalidStateError
+        val authenticatorSyncKey = authDiskSource
+            .getAuthenticatorSyncUnlockKey(userId = userId)
+            ?: return VaultUnlockResult.InvalidStateError
+        return unlockVaultForUser(
+            userId = userId,
+            initUserCryptoMethod = InitUserCryptoMethod.DecryptedKey(
+                decryptedUserKey = authenticatorSyncKey,
+            ),
+        )
+            .also {
+                if (it is VaultUnlockResult.Success) {
```

Comment:
Cool, we are kind of using this scope function the same way you could use the `onSuccess(action)` for the Kotlin`Result`?! Nice 😎

Reply:
Totally! Can't say this was my idea, I was inspired by a few other functions, but I'm digging it.